src/benchq/resource_estimation/decoder_resource_estimator.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+from typing import Optional
+
+from benchq.data_structures.decoder import DecoderModel
+from benchq.data_structures.hardware_architecture_models import BasicArchitectureModel
+from benchq.data_structures.resource_info import DecoderInfo
+
+
+def get_decoder_info(
+    hw_model,
+    decoder_model: Optional[DecoderModel],
+    code_distance: int,
+    space_time_volume: float,
+    n_logical_qubits: int,
+):
```

Comment:
Would be good to add the return type here: `Optional[DecoderInfo]`.

Reply:
Done in most recent comment!